moveit_planners/ompl/ompl_interface/src/detail/state_validity_checker.cpp

```diff
@@ -194,4 +196,112 @@ double ompl_interface::StateValidityChecker::clearance(const ompl::base::State*
   collision_detection::CollisionResult res;
   planning_context_->getPlanningScene()->checkCollision(collision_request_with_distance_, res, *robot_state);
   return res.collision ? 0.0 : (res.distance < 0.0 ? std::numeric_limits<double>::infinity() : res.distance);
+}
+
+/*******************************************
+ * Constrained Planning StateValidityChecker
+ * *****************************************/
+
+bool ompl_interface::ConstrainedPlanningStateValidityChecker::isValid(const ompl::base::State* wrapped_state,
+                                                                      bool verbose) const
+{
+  // Unwrap the state from a ConstrainedStateSpace::StateType
+  auto state = wrapped_state->as<ompl::base::ConstrainedStateSpace::StateType>()->getState();
+
+  // Use cached validity if it is available
+  if (state->as<ModelBasedStateSpace::StateType>()->isValidityKnown())
+    return state->as<ModelBasedStateSpace::StateType>()->isMarkedValid();
+
+  // do not use the unwrapped state here, as satisfiesBounds expects a state of type ConstrainedStateSpace::StateType
+  if (!si_->satisfiesBounds(wrapped_state))
```

Comment:
Can we name `si_` to something more descriptive?

Reply:
It's from the OMPL source code (`ompl::base::StateValidityChecker`). I could create a pull request there, but I can imagine it would be quite a large one.

I can also add a comment above every use here, it's only used 4 times.